pkg/url/url.go

```diff
@@ -396,6 +398,36 @@ func GetValidPortNumber(componentName string, portNumber int, portList []string)
 	return portNumber, fmt.Errorf("given port %d is not exposed on given component, available ports are: %s", portNumber, strings.Trim(strings.Replace(fmt.Sprint(componentPorts), " ", ",", -1), "[]"))
 }
 
+// GetValidExposedPortNumber checks if the given exposed port number is a valid port or not
+// if exposed port is not provided, a random 5 digit valid port will be generated and returned
+func GetValidExposedPortNumber(componentName string, exposedPort int) (int, error) {
+	// exposed port number will be -1 if the user doesn't specify any port
+	if exposedPort == -1 {
+		startPort := 49152
+		endPort := 65535
+		var port int
+		// get a random available port
+		for {
+			randomPort := rand.Intn(endPort-startPort) + startPort
+			listener, err := net.Listen("tcp", ":"+strconv.Itoa(randomPort))
+			if err == nil {
+				port = randomPort
+				listener.Close()
+				break
+			}
+		}
+		return port, nil
```

Comment:
We already have a function in util for finding in utils package `HttpGetFreePort()`. Please use that here.

Reply:
Thanks for pointing this out! Updated.

Reply:
OK cool :+1: